### PR TITLE
Fix default pad value for label maps

### LINKED ISSUE #626: Pad operation modifies labels
**🐛Bug**
The Affine transformation modifies labels at the edges.

**To reproduce**

```python
import torchio as tio
import torch
a = torch.tensor([[1, 0, 0],[0, 1, 0],[0, 0, 1]]).reshape(1, 3, 3, 1)
s = tio.Subject(lab=tio.LabelMap(tensor=a))
cp = tio.RandomAffine(translation = (0, 1, 1), default_pad_value='otsu')
sb = cp(s)
print(sb.lab.data)

```

**Expected behavior**

The augmented map is now corrupted at the edges after the translation.
The default padding value should always be 0 for labels.

**Actual behavior**

At the moment the values from outside of the original data range are set to some non-zero value.
This is expected for the non-label maps, but is a corruption of data for the labels.

### PR DESCRIPTION
**Pull request by fepegar**
Fixes #626.

**Description**
If the Otsu technique is used to compute the default padding value for the affine augmentation, it should only be used for scalar images, as noted by @glupol in #626.

**Checklist**

- [x] I have read the [`CONTRIBUTING`](https://github.com/fepegar/torchio/blob/master/CONTRIBUTING.rst) docs and have a developer setup (especially important are `pre-commit`and `pytest`)
- [x] Non-breaking change (would not break existing functionality)
- [ ] Breaking change (would cause existing functionality to change)
- [x] Tests added or modified to cover the changes
- [x] Integration tests passed locally by running `pytest`
- [ ] In-line docstrings updated
- [x] Documentation updated, tested running `make html` inside the `docs/` folder
- [x] This pull request is ready to be reviewed
- [ ] If the PR is ready and there are multiple commits, I have [squashed them and force-pushed](https://www.w3docs.com/snippets/git/how-to-combine-multiple-commits-into-one-with-3-steps.html#force-pushing-commits-7)

### CODE REVIEW

**Comment by fepegar** [comment]
@all-contributors please add @glupol for bug
